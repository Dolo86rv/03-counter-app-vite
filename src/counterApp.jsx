import React, {useState} from 'react'
import PropTypes from 'prop-types'

export const CounterApp = ({value}) => {
    
    const [counter, setCounter] =useState(value)

    const handleRest = (event) => {
        setCounter(counter - 1)
    }
    const handleAdd = (event) => {
        setCounter(counter + 1)
    }

    const handleReset = (event) => {
        setCounter(value)
    }
    return (
        <div>
            <h1>Counter App</h1>
            <h2>{value}</h2>
            <h2>{ counter }</h2>
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleRest}>-1</button>
            <button aria-label="btn-reset" onClick={handleReset}>Reset</button>
        </div>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number
}

CounterApp.defaultProps = {
    value: 'No es un numero'
}