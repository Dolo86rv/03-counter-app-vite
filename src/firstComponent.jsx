import React from 'react'
import PropTypes from 'prop-types'

const newMessage = {
    message: 'Hola',
    title: 'Dolores'
}

const getNombre = (nombre) =>{
    return nombre 
}

export const FirstComponent = ({ title, subtitle, name }) => {
    
    return (
        <>
            <p>Welcome!!!</p>
            <code>{ JSON.stringify(newMessage)}</code>
            <p>{getNombre('Valentina')}</p>
            <h1 data-testid="test-title">{title}</h1>
            <p>{subtitle}</p>
            <p>{name}</p>
        </>
        
    )
}

FirstComponent.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number.isRequired,
}
FirstComponent.defaultProps = {
    title: 'No hay titulo',
    subtitle: 'No hay subtitulo',
    name: 'Dolores Rodriguez'
}