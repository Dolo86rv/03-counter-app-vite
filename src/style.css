html, body {
    background-color: aqua;
    color: black;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}