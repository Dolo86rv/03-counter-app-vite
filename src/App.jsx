import React from 'react'
import { CounterApp } from './counterApp'
import { FirstComponent } from './firstComponent'


export const App = () => {
    return (
        <>
            <FirstComponent title='Hi, I am Dolores' subtitle={123} />
            <CounterApp value={100} />
        </>
    )
}
