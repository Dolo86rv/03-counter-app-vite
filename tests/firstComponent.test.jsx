import { render } from '@testing-library/react'
import { FirstComponent } from '../src/firstComponent'

describe('Pruebas en <FirstComponent />', () => { 
    
    test('debe de hacer match con el snapshot', () => { 
        
        const title = 'Hi, I am Dolores'

        const { container } = render( <FirstComponent title={title} />)

        expect( container ).toMatchSnapshot()


    })

    test('debe de mostrar el titulo en un h1', () => { 

        const title = 'Hi, I am Dolores'
        const { container, getByText, getByTestId } = render( <FirstComponent title={title} />)
        expect(getByText(title)).toBeTruthy()

        expect(getByTestId('test-title').innerHTML).toBe(title)

        //const h1 = container.querySelector('h1')
        //expect(h1.innerHTML).toBe(title)
    })

    test('debe de mostrar el subtitulo enviado por props', () => {
        const title = 'Hi, I am Dolores'
        const subtitle = 'Soy un subtitulo'
        const { getByText } = render( 
            <FirstComponent 
                title={title}
                subtitle={subtitle} 
        />)
        expect(getByText(subtitle)).toBeTruthy()
    })
})