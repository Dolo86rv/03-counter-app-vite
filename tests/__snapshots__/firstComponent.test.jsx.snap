// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstComponent /> debe de hacer match con el snapshot 1`] = `
<div>
  <p>
    Welcome!!!
  </p>
  <code>
    {"message":"Hola","title":"Dolores"}
  </code>
  <p>
    Valentina
  </p>
  <h1
    data-testid="test-title"
  >
    Hi, I am Dolores
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Dolores Rodriguez
  </p>
</div>
`;
